ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/PlanInfo.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.plan;
+
+/**
+ * Information about an execution step plan to be passed to a {@link PlanBuilder}
+ * for use when converting an execution step plan into a physical plan.
+ */
+public interface PlanInfo {
```

Comment:
Is it weird that this interface is empty? It's unclear to me which (if any) parts of the KSPlanInfo implementation are generic enough to warrant being moved into the interface.

Reply:
I think the abstract builder pattern makes sense for `PlanBuilder`, but less so for `PlanInfoExtractor` because we may want different `PlanBuilder`s if we decide to implement a non-KS physical plan (ignoring for a sec that `KStreamHolder` isn't actually generic enough to support non-KS implementations yet). The `PlanInfo`, however, should make sense to any corresponding physical plan we decide to build (it's an analysis of the steps and has nothing to do with Kafka Streams specifically). All we have to do is rename "repartition" to "keyChange" and this change becomes entirely generic. 

the tl;dr is that I don't think we need to have this be an interface (and I don't think we need an interface for `PlanInfoExtractor` either (though see my general comment about a different approach to this)

Reply:
I agree with your reasoning regarding the visitor pattern for `PlanBuilder`. My thinking in abstracting `PlanInfoExtractor` was that different plan builder implementations will likely want different information from the plan. For example, a non-Kafka-Streams implementation might not care about repartitions.

This is all very up in the clouds, though, since having a non-Kafka-Streams implementation is very far off. If you prefer nixing the interface I'm happy to do so.

Reply:
I'd rather nix the interface 😂 fewer lines of code to refactor when we inevitably do